src/app_service/service/message/async_tasks.nim

```diff
@@ -6,6 +6,20 @@ import ../../../backend/chat as status_go_chat
 
 import ../../../app/core/custom_urls/urls_manager
 
+import dto/seen_unseen_messages
+
+proc getCountAndCountWithMentionsFromResponse(chatId: string, seenAndUnseenMessagesBatch: JsonNode): (int, int) =
+  var count: int
+  var countWithMentions: int
+
+  if seenAndUnseenMessagesBatch.len > 0:
```

Comment:
It would consider refactoring this function because it has too many branches.
You can use early exit with the break to act like a precondition. And I believe it's a hidden trap to use break in the middle of the block.

Reply:
I don't think it's worth changing this feature, but anyway it's easier fix than debate :)